### ISSUE TITLE
Error detected while processing /usr/share/vim/vim80/defaults.vim when invoked as vi

### ISSUE THREAD

**Issue by crd477** [vim/vim#1604]
I am encountering the following errors when invoking vim as /usr/bin/vi:

```
$ /usr/bin/vi a
Error detected while processing /usr/share/vim/vim80/defaults.vim:
line  100:
E10: \ should be followed by /, ? or &
line  101:
E10: \ should be followed by /, ? or &
line  102:
E10: \ should be followed by /, ? or &
line  114:
E10: \ should be followed by /, ? or &
Press ENTER or type command to continue
```

Pressing enter allows me to continue without any apparent ill effects but the error reporting is new.  The problem appears to be to have been introduced between versions 8.0.0329 and 8.0.0494.  Version 8.0.0525 also exhibits the problem.

I am using vim on Cygwin with the mintty terminal and `TERM=xterm`.  I reported this issue to the Cygwin package maintainer and he confirmed this error when the `vim-common` package is installed and suggested that I report it upstream.

Here is the version output from `:version` on my vim installation:
```
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Mar 29 2017 17:57:07)
Included patches: 1-525
Modified by <cygwin@cygwin.com>
Compiled by <cygwin@cygwin.com>
Small version without GUI.  Features included (+) or not (-):
+acl             -ebcdic          -listcmds        -printer         -title
-arabic          -emacs_tags      -localmap        -profile         -toolbar
-autocmd         -eval            -lua             -python          -user_commands
-balloon_eval    +ex_extra        -menu            -python3         +vertsplit
-browse          -extra_search    -mksession       -quickfix        -virtualedit
+builtin_terms   -farsi           -modify_fname    -reltime         +visual
-byte_offset     -file_in_path    -mouse           -rightleft       -visualextra
-channel         -find_in_path    -mouse_dec       -ruby            -viminfo
-cindent         -float           -mouse_gpm       -scrollbind      -vreplace
-clientserver    -folding         -mouse_jsbterm   -signs           +wildignore
-clipboard       -footer          -mouse_netterm   -smartindent     -wildmenu
-cmdline_compl   +fork()          -mouse_sgr       -startuptime     +windows
+cmdline_hist    -gettext         -mouse_sysmouse  -statusline      +writebackup
-cmdline_info    -hangul_input    -mouse_urxvt     -sun_workshop    -X11
-comments        +iconv           -mouse_xterm     -syntax          -xfontset
-conceal         -insert_expand   +multi_byte      +tag_binary      -xim
-cryptv          -job             -multi_lang      -tag_old_static  -xpm
-cscope          +jumplist        -mzscheme        -tag_any_white   -xsmp
-cursorbind      -keymap          -netbeans_intg   -tcl             -xterm_clipboard
-cursorshape     -lambda          -num64           -termguicolors   -xterm_save
-dialog          -langmap         +packages        +terminfo
-diff            -libcall         -path_extra      -termresponse
-digraphs        -linebreak       -perl            -textobjects
-dnd             -lispindent      -persistent_undo -timers
   system vimrc file: "/etc/virc"
     user vimrc file: "$HOME/.virc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
  fall-back for $VIM: "/etc"
 f-b for $VIMRUNTIME: "/usr/share/vim/vim80"
Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H     -ggdb -O2 -pipe -Wimplicit-function-declaration
-fdebug-prefix-map=/usr/src/ports/vim/vim-8.0.0525-1.x86_64/build=/usr/src/debug/vim-8.0.0525-1 -fde
bug-prefix-map=/usr/src/ports/vim/vim-8.0.0525-1.x86_64/src/vim-8.0.0525=/usr/src/debug/vim-8.0.0525
-1 -fstack-protector-strong -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1
Linking: gcc   -fstack-protector-strong -L/usr/local/lib -Wl,--as-needed -o vim.exe        -lm  -lnc
ursesw -liconv -lacl -lattr -lintl
```

**Comment by chrisbra** [comment]
this is caused by patch 463, that changes how to set `nocompatible`. Unfortunately, your Vim does not have eval feature, so it won't be set and therefore you see those errors later.

**Comment by yselkowitz** [comment]
This is coming from Cygwin's /bin/vi, which is vim configured --with-features=small, and packaged in the Base installation without a dependency on vim-common ($VIMRUNTIME).  When vim-common is not installed, it works fine, but if vim-common is installed (either by vim or gvim, both built --with-features=huge and the interpreters enabled), this error message occurs.

What should be done so that this vi is not "allergic" to defaults.vim when it is installed.  Can it be configured to ignore $VIMRUNTIME, or does defaults.vim need to be patched to be compatible with a --with-features=small build?

**Comment by chrisbra** [comment]
I think the simplest solution would be this fix to defaults.vim:
https://github.com/chrisbra/vim/commit/db175c7334828433f4175af8fb2726fe1513cc74

However I don't like how this duplicates certain options values, don't have a better idea currently. Perhaps Bram has a simpler solution.

**Comment by brammool** [comment]
It does look like the most robust solution.  To avoid duplication we can have "if 1" a few more times.

**Comment by chrisbra** [comment]
fixed as of [8.0.534](https://github.com/vim/vim/releases/tag/v8.0.0534)
